packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -434,13 +646,22 @@ export class Pipeline extends PipelineBase {
       assumedBy: new iam.ServicePrincipal('codepipeline.amazonaws.com'),
     });
 
+    if (props.pipelineType === undefined) {
+      Annotations.of(this).addWarningV2('@aws-cdk/aws-codepipeline:unspecifiedPipelineType', 'V1 pipeline type is implicitly selected when `pipelineType` is not set. If you want to use V2 type, set `PipelineType.V2`.');
```

Comment:
Musings (for a separate PR): should we add a feature flag that changes the default pipeline to be v2?

Reply:
OK, I will change the default type to V2 under feature flag.

Reply:
https://github.com/aws/aws-cdk/pull/28538/commits/403d7fa92173fb1fef3628396ba3fb67cb49758e